### PR TITLE
Example for 1 ingress routing to many applications

### PR DESCRIPTION
**Pull request by Blizter**
Hello, 

The examples show how to generate a manifest for one ingress per application. I didn't find any example anywhere on how to define a one-to-many ingress, which is more cost-efficient in some cases. 

The modified block between 'examples/ingress-one-to-many.jsonnet' and 'examples/ingress.jsonnet' is able to generate the ingress manifest correctly on my end with another ingress controller, a manifest based on the example is currently deployed. 

I took care of keeping the same annotations as well naming convention from an example file to another.

Hope this helps people looking for a more cost-efficient way to expose many applications with only one ingress. 

Please, advise modification in documentation (i.e. ; `docs/exposing-prometheus-alertmanager-grafana-ingress.md`) 
if needed, I'll gladly make them :)

### CODE REVIEW

**Comment by brancz** [comment]
We're actually in the process of removing ksonnet from this repo: https://github.com/prometheus-operator/kube-prometheus/issues/591#issuecomment-722976567

Could you change this PR to not use ksonnet? Then lgtm! :)

**Comment by Blizter** [comment]
> We're actually in the process of removing ksonnet from this repo: [#591 (comment)](https://github.com/prometheus-operator/kube-prometheus/issues/591#issuecomment-722976567)
> 
> Could you change this PR to not use ksonnet? Then lgtm! :)

Yes, I can try to do that. On top of looking at the PRs removing ksonnet, should I use [this file](https://github.com/prometheus-operator/kube-prometheus/blob/master/examples/ingress.jsonnet) as a reference to remove ksonnet?

**Comment by Blizter** [comment]
@brancz modified the example to remove ksonnet :)

File: examples/ingress-one-to-many.jsonnet [hunk 1]
```diff
@@ -0,0 +1,109 @@
+local ingress(name, namespace, rules) = {
```
Comment (brancz) [inline examples/ingress-one-to-many.jsonnet:1]: I think it would be easier for beginners to understand if we didn't use a function here.

**Review by brancz — COMMENTED** [review]
Just a small thing for new users to understand this easier, otherwise really awesome, thank you!

File: examples/ingress-one-to-many.jsonnet (hunk 1 above)
Comment (Blizter) [reply examples/ingress-one-to-many.jsonnet:1]: Yes, sure! Where can I put this function?

Comment (brancz) [reply examples/ingress-one-to-many.jsonnet:1]: Just no function, I think we should just in-line the entire ingress object.

Comment (Blizter) [reply examples/ingress-one-to-many.jsonnet:1]: Since so much time has passed, should I close this PR?

**Comment by paulfantom** [comment]
Closing as we merged it with #813. Please reopen if you feel this is still needed.

**Pull request closed without merging**